content/en/profiler/connect_traces_and_profiles.md

```diff
@@ -322,11 +322,23 @@ With endpoint profiling you can:
 
 {{< img src="profiler/endpoint_agg.mp4" alt="Troubleshooting a slow endpoint by using endpoint aggregation" video=true >}}
 
-### Track the endpoints that consume the most resources
+### Surface code that impacted your production latency
+
+In APM Service page, use `Profiling` tab to correlate a latency/throughput change to a code performance change.
+
+In this example, you can see how latency is linked to a lock contention increase on `/GET train` caused by the following line of code
+
+```java
+Thread.sleep(DELAY_BY.minus(elapsed).toMillis());
+```
+
+{{< img src="profiler/apm_service_page_pivot_to_contention_comparison.mp4" alt="Pivoting from APM service page to Profiling comparison page to find the line of code causing latency" video=true >}}
+
+### Track endpoints that consume the most resources
 
 It is valuable to track top endpoints that are consuming valuable resources such as CPU and wall time. The list can help you identify if your endpoints have regressed or if you have newly introduced endpoints that are consuming drastically more resources, slowing down your overall service.
 
-The following image shows that `GET /store_history` is periodically impacting this service by consuming 20% of its CPU:
+The following image shows that `GET /store_history` is periodically impacting this service by consuming 20% of its CPU and 50% of its allocated memory:
```

Comment:
Could you change the name of the new image? There's a caching issue when we use the same image name, the new image won't show up.

Reply:
Done in https://github.com/DataDog/documentation/pull/22475/commits/d812efe47fcbab3f60e88a8dad22fb5a63af1406